packages/babel-types/src/modifications/typescript/removeTypeDuplicates.ts

```diff
@@ -47,7 +55,27 @@ export default function removeTypeDuplicates(
       continue;
     }
 
-    // TODO: add generic types
+    // todo: support merging tuples: number[]
+    if (isTSTypeReference(node) && node.typeParameters) {
+      const name = getQualifiedName(node.typeName);
+
+      if (generics[name]) {
```

Comment:
We should make sure that `generics` has a null prototype, otherwise this risks reading things from `Object.prototype`.